src/sentry/runner/__init__.py

```diff
@@ -161,4 +165,22 @@ def call_command(name, obj=None, **kwargs):
 
 
 def main():
-    cli(prog_name=get_prog(), obj={}, max_content_width=100)
+    try:
+        if os.environ.get("SENTRY_DEVENV_DSN"):
+            logger.warning(
+                "The Sentry runner will report development issues to Sentry.io."
+                "Use SENTRY_DEVENV_NO_REPORT to avoid reporting issues."
+            )
+        cli(prog_name=get_prog(), obj={}, max_content_width=100)
+    except Exception as e:
+        if os.environ.get("SENTRY_DEVENV_DSN"):
+            # This reports to the project sentry-dev-env
+            sentry_sdk.init(
```

Comment:
put this in a with-stmt:

```
with init():
    # capture_exception etc
```

then the old client will be restored at the end

Reply:
actually it won't (it just blocks until the error has been sent) but I'd still recommend